src/convert.jl

```diff
@@ -21,6 +22,13 @@ Convert Trixi-generated output files to VTK files (VTU or VTI).
                A value of `0` (zero) uses the number of nodes in the DG elements.
 - `save_celldata`: Boolean value to determine if cell-based data should be saved.
                    (default: `true`)
+- `reinterpolate`: Boolean value to determine if data should be reinterpolated
+                   onto uniform points. When `false` the raw data at the compute nodes
+                   is copied into the appropriate format.
+                   (default: `true`)
+- `data_is_uniform`: Boolean to indicate if the data to be converted is from a finite difference
+                     method on a uniform grid of points.
+                     (default: `false`)
```

Comment:
The situation `reinterpolate=false` and `data_is_uniform=false` would occur if one wants to plot the solution on the raw DGSEM points. The `reinterpolate=true` and `data_is_uniform=true` should never occur in partice (and be avoided in general) because constructing high-degree interpolations on uniform nodes is prone to Runge phenomena.

It would be nice to have just a `reinterpolate` switch to turn ON/OFF. The reason I needed two was to set the correct nodes. When DGSEM does not reinterpolate then the `nodes` need to be the LGL nodes whereas when FD does not reinterpolate the nodes are uniform. If the solver type was stored in the `.h5` file then this `nodes` check could be done on `DGSEM` and `FDSBP` respectively.

Reply:
IMHO we should then aim to include the solver type in mesh file. We didn't before, since we essentially had only a single scheme or at least a single way to interpret the results.

I would understand if you don't feel like doing this in Trixi right now; however, in that case it would be good to create an issue which references this discussion.

Reply:
> Do we store the `nodes` in the HDF5 file? There are also some FDSBP operators with non-uniform grid near the boundaries.

No. In the TreeMesh, the nodes can be reconstructed from the cell center coordinates + cell length. For P4estMesh, I believe we store the tree node coordinates, from which the node locations can be reconstructed.

Reply:
If we really want to support arbitrary (FD) SBP methods, we need to store their local 1D solution nodes of their "reference element.

Reply:
> I would understand if you don't feel like doing this in Trixi right now; however, in that case it would be good to create an issue which references this discussion.

You are right, I do not feel like doing this right now :) but I can open issues in Trixi2Vtk and in Trixi that reference this conversation. If we alter the `.h5` output to include the solver type *and* the nodes then the logic of the `trixi2vtk` function could simplify considerably. It also would allow reinterpolation of FD methods with nonuniform boundary closures (a type of method I completely forgot existed).

Reply:
Opened an [issue]((https://github.com/trixi-framework/Trixi.jl/issues/1300)) in Trixi and a corresponding [issue](https://github.com/trixi-framework/Trixi2Vtk.jl/issues/62) in Trixi2Vtk.

Reply:
Great, thanks a lot! /c